errors.go

```diff
@@ -33,3 +33,9 @@ const ErrRejected = errorType("response rejected")
 
 // ErrUnsupported indicates an operation is not supported by the transport protocol
 const ErrUnsupported = errorType("unsupported")
+
+// ErrDisconnected indicates the other peer may have hung up and you should try restarting the channel.
```

Comment:
@hannahhoward For my own curiosity:

What's the advantage of defining:

```
type errorType string

func (e errorType) Error() string {
	return string(e)
}
```
rather than using `errors.New()`

Reply:
it allows you to use `const` instead of `var` -- there is a danger of `var` being changed -- technically anyone using the module can do so.